lib/Service/DocumentService.php

```diff
@@ -182,59 +192,68 @@ public function get($documentId) {
 	 * @param $version
 	 * @return array
 	 * @throws DoesNotExistException
-	 * @throws VersionMismatchException
+	 * @throws InvalidArgumentException
 	 */
 	public function addStep($documentId, $sessionId, $steps, $version): array {
-		$document = null;
-		$oldVersion = null;
-
-		try {
-			$this->lockingProvider->acquireLock('document-push-lock-' . $documentId, ILockingProvider::LOCK_EXCLUSIVE);
-		} catch (LockedException $e) {
-			throw new VersionMismatchException('Version does not match');
+		$stepsToInsert = [];
+		$querySteps = [];
+		$getStepsSinceVersion = null;
+		$newVersion = $version;
+		foreach ($steps as $step) {
+			if ($step > "AAE") {
+				array_push($stepsToInsert, $step);
+			} else {
+				array_push($querySteps, $step);
+			}
 		}
+		if (sizeof($stepsToInsert) > 0) {
+			$newVersion = $this->insertSteps($documentId, $sessionId, $stepsToInsert, $version);
+		}
+		// If there were any queries in the steps send the entire history
+		$getStepsSinceVersion = sizeof($querySteps) > 0 ? 0 : $version;
```

Comment:
```suggestion
		$getStepsSinceVersion = count($querySteps) > 0 ? 0 : $version;
```

I'd prefer using count as sizeof has different meanings in other languages so this may lead to confusion.